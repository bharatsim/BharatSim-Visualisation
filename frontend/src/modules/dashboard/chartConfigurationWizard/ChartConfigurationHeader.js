import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

import { Typography } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import { Close } from '@material-ui/icons';
import chartConfigs from '../../../config/chartConfigs';
import IconButton from '../../../uiComponent/IconButton';

const useStyles = makeStyles((theme) => {
  return {
    imageContainer: {
      height: theme.spacing(8),
      width: theme.spacing(8),
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: theme.colors.primaryColorScale['50'],
      borderColor: theme.palette.grey['100'],
      border: '1px solid',
      marginRight: theme.spacing(3),
      borderRadius: theme.spacing(1),
    },
    image: {
      width: theme.spacing(6),
    },
    chartName: {
      color: theme.palette.text.secondary,
    },
  };
});

function ChartConfigurationHeader({ closeModal, chart, activeStep }) {
  const classes = useStyles();
  return (
    <Box pl={2} display="flex" alignItems="center" justifyContent="space-between">
      <Box display="flex" alignItems="center">
        <Typography variant="h6">Chart Configuration Wizard</Typography>
        {chart && activeStep !== 0 && (
          <Box px={6} display="flex" alignItems="center">
            <Box className={classes.imageContainer}>
              <img src={chartConfigs[chart].icon} alt={chart} className={classes.image} />
            </Box>
            <Typography variant="body2" classes={{ root: classes.chartName }}>
              {chartConfigs[chart].label}
            </Typography>
          </Box>
        )}
      </Box>
      <IconButton onClick={closeModal} data-testid="close-wizard">
        <Close />
      </IconButton>
    </Box>
  );
}

ChartConfigurationHeader.defaultProps = {
  chart: '',
};

ChartConfigurationHeader.propTypes = {
  activeStep: PropTypes.number.isRequired,
  chart: PropTypes.string,
  closeModal: PropTypes.func.isRequired,
};

export default ChartConfigurationHeader;
