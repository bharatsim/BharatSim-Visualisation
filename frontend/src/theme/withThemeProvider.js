import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
