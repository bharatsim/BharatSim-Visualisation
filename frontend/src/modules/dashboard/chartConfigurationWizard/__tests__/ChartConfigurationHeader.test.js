import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import ChartConfigurationHeader from '../ChartConfigurationHeader';
import withThemeProvider from '../../../../theme/withThemeProvider';

describe('Chart configuration header', () => {
  const ChartConfigurationHeaderWithTheme = withThemeProvider(ChartConfigurationHeader);
  it('should match snapshot without chart name', () => {
    const { container } = render(
      <ChartConfigurationHeaderWithTheme closeModal={jest.fn()} chart="" activeStep={0} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should call close wizard on click of close icon', () => {
    const closeModal = jest.fn();
    const { getByTestId } = render(
      <ChartConfigurationHeaderWithTheme
        closeModal={closeModal}
        chart="lineChart"
        activeStep={0}
      />,
    );

    const closeIcon = getByTestId('close-wizard');

    fireEvent.click(closeIcon);

    expect(closeModal).toHaveBeenCalled();
  });

  it('should show chart name and icon if chart is selected and active step is not in step one', () => {
    const closeModal = jest.fn();
    const { getByText, getByAltText } = render(
      <ChartConfigurationHeaderWithTheme
        closeModal={closeModal}
        chart="lineChart"
        activeStep={1}
      />,
    );

    expect(getByText('Line Chart')).toBeInTheDocument();
    expect(getByAltText('lineChart')).toBeInTheDocument();
  });

  it('should not show chart name and icon if chart is not selected', () => {
    const closeModal = jest.fn();
    const { queryByText } = render(
      <ChartConfigurationHeaderWithTheme closeModal={closeModal} chart="" activeStep={1} />,
    );

    expect(queryByText('line chart')).toBeNull();
  });

  it('should not show chart name and icon if active step is zero', () => {
    const closeModal = jest.fn();
    const { queryByText } = render(
      <ChartConfigurationHeaderWithTheme
        closeModal={closeModal}
        chart="lineChart"
        activeStep={0}
      />,
    );

    expect(queryByText('line chart')).toBeNull();
  });
});
