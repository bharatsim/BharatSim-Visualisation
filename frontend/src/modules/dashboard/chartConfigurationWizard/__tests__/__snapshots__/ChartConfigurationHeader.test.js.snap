// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart configuration header should match snapshot without chart name 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-4"
  >
    <div
      class="MuiBox-root MuiBox-root-5"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Chart Configuration Wizard
      </h6>
      
    </div>
    <button
      class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-6"
      data-testid="close-wizard"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
